activerecord/lib/active_record/attribute_methods.rb

```diff
@@ -196,6 +236,20 @@ def column_for_attribute(name)
           ConnectionAdapters::NullColumn.new(name)
         end
       end
+
+      private
+
+        # An Array of String attribute names and aliases for accessing those
+        # attributes.
+        #
+        #   class Person < ActiveRecord::Base
+        #   end
+        #
+        #   Person.attribute_names_and_aliases
+        #   # => ["id", "created_at", "updated_at", "name", "age"]
+        def attribute_names_and_aliases
+          @attribute_names_and_aliases ||= attribute_names + attribute_aliases.keys
```

Comment:
I guess it goes in `reload_schema_from_cache`.. though that implies two modules care about this ivar.

Or maybe I'm thinking of the wrong sort of reset: this needs to be cleared when a new attribute alias is defined, for example.

@sgrif may have some thoughts.